meta-resin-common/recipes-support/resin-vars/resin-vars/os-networkmanagerconf

```diff
@@ -0,0 +1,44 @@
+#!/bin/bash
+
+. /usr/sbin/resin-vars
+
+keys=$(jq -r 'select(.os != null) | select(.os.networkManagerConf != null) | .os.networkManagerConf | keys[] ' "$CONFIG_PATH")
+lscmd="ls"
+
+if [ ! -z "$keys" ]; then
+	echo "Found custom config fragments in $CONFIG_PATH"
+	while read -r key; do
+		echo "Found custom config fragment: $key"
+		lscmd="$lscmd -I \"$key.conf\""
+		filter="'.os.networkManagerConf.\"$key\"'"
+		cmd="jq -r $filter $CONFIG_PATH"
+		conf=$(eval "$cmd")
+		if [ -f "/etc/NetworkManager/conf.d/$key.conf" ] && [ "$(cat "/etc/NetworkManager/conf.d/$key.conf")" = "$conf" ]; then
+			echo "/etc/NetworkManager/conf.d/$key.conf already exists"
+			continue
+		else
+			echo "New config fragment or modified fragment detected. Writing /etc/NetworkManager/conf.d/$key.conf"
+			echo "$conf" > "/etc/NetworkManager/conf.d/$key.conf"
+		fi
+	done <<< "$keys"
+
+	if [ ! -f /etc/NetworkManager/conf.d/README ]; then
+		echo "Files here are auto-generated by os-networkmanagerconf. Extra files will be removed" >> /etc/NetworkManager/conf.d/README
+	fi
+
+	lscmd="$lscmd /etc/NetworkManager/conf.d"
+	extra_conf=$(eval "$lscmd")
+	while read -r extra_conf; do
+		if [ "$extra_conf" != "README" ]; then
+			echo "Removing extra conf: $extra_conf"
+			rm "/etc/NetworkManager/conf.d/$extra_conf"
+		fi
+	done <<< "$extra_conf"
+
+	echo "All config fragments from config.json present in /etc/NetworkManager/conf.d"
+else
+	echo "No custom config fragments found in $CONFIG_PATH"
+	if [ "$(ls -A /etc/NetworkManager/conf.d -I README)" ]; then
```

Comment:
Why do you ignore README when you actually remove it afterwards?

Reply:
The remove should also ignore README. But that uselessly complicates the logic. I kept it simple and removed it. It'll regenerate next time any conf fragment is added in config.json